masonry/src/widgets/button.rs

```diff
@@ -14,71 +14,50 @@ use vello::peniko::Color;
 
 use crate::core::keyboard::{Key, NamedKey};
 use crate::core::{
-    AccessCtx, AccessEvent, ArcStr, BoxConstraints, ChildrenIds, EventCtx, LayoutCtx, NewWidget,
-    PaintCtx, PointerEvent, PropertiesMut, PropertiesRef, RegisterCtx, TextEvent, Update,
-    UpdateCtx, Widget, WidgetId, WidgetMut, WidgetPod,
+    AccessCtx, AccessEvent, BoxConstraints, ChildrenIds, EventCtx, LayoutCtx, NewWidget, PaintCtx,
+    PointerEvent, PropertiesMut, PropertiesRef, RegisterCtx, TextEvent, Update, UpdateCtx, Widget,
+    WidgetId, WidgetMut, WidgetPod,
 };
 use crate::properties::{
     ActiveBackground, Background, BorderColor, BorderWidth, BoxShadow, CornerRadius,
     DisabledBackground, HoveredBorderColor, Padding,
 };
 use crate::theme;
 use crate::util::{fill, include_screenshot, stroke};
-use crate::widgets::Label;
 
 /// A button with a text label.
 ///
 /// Emits [`ButtonPress`] when pressed.
 ///
 #[doc = include_screenshot!("button_hello.png", "Button with text label.")]
 pub struct Button {
-    label: WidgetPod<Label>,
+    child: WidgetPod<dyn Widget>,
 }
 
 // --- MARK: BUILDERS
 impl Button {
-    /// Create a new button with a text label.
+    /// Create a new button with a child widget.
     ///
     /// # Examples
     ///
     /// ```
-    /// use masonry::widgets::Button;
-    ///
-    /// let button = Button::new("Increment");
-    /// ```
-    pub fn new(text: impl Into<ArcStr>) -> Self {
-        Self::from_label(Label::new(text).with_auto_id())
-    }
```

Comment:
I think it'd make sense to keep that constructor as the "easy default".

Reply:
We can do that, but not sure how would one edit the properties of the `Label` tho?